server/server/src/main/kotlin/org/jetbrains/bsp/bazel/server/sync/languages/java/JavaLanguagePlugin.kt

```diff
@@ -53,7 +56,23 @@ class JavaLanguagePlugin(
   private fun getJdk(): Jdk = jdk ?: throw RuntimeException("Failed to resolve JDK for project")
 
   override fun dependencySources(targetInfo: TargetInfo, dependencyGraph: DependencyGraph): Set<URI> =
-    emptySet() // Provided via workspace/libraries
+    targetInfo.getJvmTargetInfoOrNull()?.run {
```

Comment:
and I'd do this calculation conditional as well - it's quite heavy calculation executed for each target, what can add quite some overhead for us (we use a separate mechanism for it, because this one is too slow for us).

By "conditional" everywhere I meant that it's possible to check what rules are imported (org/jetbrains/bsp/bazel/server/sync/ProjectResolver.kt:69), it's a list of all rules in the workspace, and then you can use this list to check if the fork rules are loaded

Reply:
Ok, so on this one I have a question. As far as I can tell, the `workspace/libraries` command isn't a part of the [build server protocol](https://build-server-protocol.github.io/docs/specification) that I can see, and more practically speaking, metals lsp doesn't seem to consume it. All that is to say, I enabled this more for metals support than for rules_scala_annex support. 

I'm happy to do what you recommended just to make my use case work, but is it possible we could make this something configurable instead? I see there are some FeatureFlags in the repo, would that be a viable place to add a toggle for something like this?

Reply:
> override it only in our client

I'm not sure how to do that. Does that need to happen in another repo?